ui/package.json

```
@@ -16,8 +16,9 @@
     "test:watch": "jest --watch",
     "test:debug": "node --inspect-brk node_modules/.bin/jest --runInBand --watch",
     "lint": "yarn run eslint; yarn run tsc",
-    "test:e2e": "CYPRESS_baseUrl=http://localhost:8888 cypress run --browser chrome --headless --reporter junit --reporter-options 'mochaFile=cypress/results/results-[hash].xml'",
-    "test:e2e:headed": "CYPRESS_baseUrl=http://localhost:8888 cypress open --browser chrome",
+    "test:e2e": "nc -z ${OAUTH2_HOSTNAME:=localhost} ${OAUTH2_PORT:=8087} && CYPRESS_oauth2ServerURL=http://${OAUTH2_HOSTNAME}:${OAUTH2_PORT} cypress run --browser chrome --headless --reporter junit --reporter-options 'mochaFile=cypress/results/results-[hash].xml' || echo 'Unable to contact OAuth2 Server. Type \"yarn test:oauth-mock\" to start it.'",
+    "test:e2e:headed": "nc -z ${OAUTH2_HOSTNAME:-localhost} ${OAUTH2_PORT:-8087} && CYPRESS_oauth2ServerURL=http://${OAUTH2_HOSTNAME:-localhost}:${OAUTH2_PORT:-8087} cypress open --browser chrome || echo 'Unable to contact OAuth2 Server. Type \"yarn test:oauth-mock\" to start it.'",
+    "test:oauth-mock": "yarn --cwd ../etc/oauth2-server-mock start",
```

it would be better to make this target standalone (so it just works OOTB) with:
```suggestion
    "test:oauth-mock": "yarn --cwd ../etc/oauth2-server-mock install && yarn --cwd ../etc/oauth2-server-mock start",
```

---

Script adjusted.